### ISSUE TITLE
Nested rattan in `Compatible` mode

### ISSUE THREAD

**Issue by Centaurus99** [stack-rs/rattan#29]

**Comment by Centaurus99** [comment]
In `Compatible` mode nesting, there are currently two main issues:

1. **IP Address Conflict**: Rattan only detects used IP addresses within the netns where it is started. As a result, when one Rattan instance is started inside another, it does not detect IP address conflicts, causing the inner and outer Rattans to use the same address.
2. **Missing Routes**: The inner uses the default route to reach the outer, which is not an issue. However, the outer does not know the IP of the nested inner and cannot add the corresponding route, so it cannot reach the nested inner.

**Comment by BobAnkh** [comment]
> Before implementing this feature, I suggest adding a check in the rattan CLI to detect whether it is already inside a rattan, and explicitly preventing the creation of nested rattans. [@BobAnkh](https://github.com/BobAnkh)

I agree with you. So, shall we detect the RATTAN_BASE or shall we detect our current NS? You can add it if you have time.
